src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -619,35 +619,38 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         }
 
         const auto modifiers = _GetPressedModifierKeys();
-        const auto vkey = static_cast<WORD>(e.OriginalKey());
 
+        // AltGr key combinations don't always contain any meaningful,
+        // pretranslated unicode character during WM_KEYDOWN.
+        // E.g. on a German keyboard AltGr+Q should result in a "@" character,
+        // but actually results in "Q" with Alt and Ctrl modifier states.
+        // By returning false though, we can abort handling this WM_KEYDOWN
+        // event and let the WM_CHAR handler kick in, which will be
+        // provided with an appropriate unicode character.

```

Comment:
I'd maybe also mention in this comment something like the following:
```suggestion
        // provided with an appropriate unicode character.
        // GH#2235: Make sure to handle AltGr before trying keybindings, so 
        // Ctrl+Alt keybindings won't eat an AltGr keypress
```

Reply:
FYI: I had to rebase your suggestion due to formatting issues.